centaur/src/main/scala/centaur/api/DaemonizedDefaultThreadFactory.scala

```diff
@@ -14,8 +14,7 @@ import java.util.concurrent.atomic.AtomicInteger
   * > main thread).
   */
 object DaemonizedDefaultThreadFactory extends ThreadFactory {
-  private val s = System.getSecurityManager
-  private val group = if (s != null) s.getThreadGroup else Thread.currentThread.getThreadGroup
```

Comment:
`SecurityManager` is deprecated in 17. I honestly don't know what its role is here, but it certainly seems like the logic is just fine with the possibility of it being `null`.